### PR TITLE
#312 additional properties

### LINKED ISSUE #312: Add v2 additional property data
The following properties also need to be imported and displayed:

- [ ] NZULM
- [ ] EML product
- [ ] EML item
- [ ] UNSPSC
- [ ] DDD
- [ ] WHO ATC

## Expected Behaviour

All but DDD & ATC are available in an excel sheet. Those two will need to be imported from previous JSON dataset.
Should be added to the import routine and `product.csv` and displayed on the site

## Current Behaviour

Properties are displayed on current (v1) site. They are not in the v2 site.

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #312 

## Description
Decided to leave DDD and WHO ATC codes out until they have been checked

Updated the csv from the current working version of the spreadsheet. This now includes NZULM, EML and UNSPSC codes.

Added support for these in the import routine and tested.

### Documentation:

Tick below if one of the following applies:

1. Documentation has been added to cover changes introduced by this PR.
2. This PR requires no changes to documentation.

- [ ] Documentation OK.

### Tests:

Tick below if one of the following applies:

1. Tests have been added to cover changes introduced by this PR.
2. This PR requires no new tests.

- [ ] Tests OK.

### CODE REVIEW

**Review by katherine-sussol — APPROVED** [review]
👍  once #310 is merged this one looks good

File: unified-codes/tools/scripts/dgraph/data/v2/products.csv [hunk 1]
```diff
@@ -1,1099 +1,1185 @@
...
-Whole Blood,,,Parenteral,Liquid,,,450mL,,,,,,,,,,,,,,,,
-Xylometazoline,,,Nasal,Spray,,0.05%,10mL,,,,933f3f00,6f72387f,d1bf7b00,ba0c5200,,6766a4bf,,,,,39841,,
-Yellow Fever Vaccine,,,Parenteral,Powder for reconstitution,Intramuscular,> 1000IU,0.5mL,Vial,,,,,,,,,,,,,,,
-Zidovudine,"ZDV, AZT",,Oral,Capsule,,100mg,,,,,933f3f00,02ac6c47,274d5c00,c626af00,,6773e4bf,,,,,11413,,
-Zidovudine,"ZDV, AZT",,Oral,Capsule,,250mg,,,,,933f3f00,02ac6c47,274d5c00,c626af00,,6776f4bf,,,,,11413,,
-Zidovudine,"ZDV, AZT",,Oral,Solution,,10mg per mL,200mL,,,,933f3f00,02ac6c47,274d5c00,aaa6b600,,,6770b4bf,,,,11413,,
-Zidovudine,"ZDV, AZT",,Oral,Tablet,,300mg,,,,,933f3f00,02ac6c47,274d5c00,ea02a100,,677a04bf,,,,,11413,,
-Zidovudine,"ZDV, AZT",,Parenteral,Solution,,10mg per mL,20mL,,,,933f3f00,02ac6c47,e541cd00,,,,676cd4bf,,,,11413,,
-Zinc Sulfate,,,Oral,Tablet,Dispersible,20mg,,,,,933f3f00,5d4cf661,d55ae000,c70e2300,,677cf4bf,,,,,39954,,
+﻿Product ,Product Synonym,Combination,Route,Dose Form,Dose Qualification,Strength,Unit of Presentation,Immediate Packaging,Pack Size,Outer Packaging,Brand,UC1,"UC2
```
[the remark this answers is not in the document]
Comment (wlthomson) [reply unified-codes/tools/scripts/dgraph/data/v2/products.csv:1]: giphy

File: unified-codes/tools/scripts/dgraph/migration/v2/DataLoader.ts [hunk 2]
```diff
@@ -93,15 +93,14 @@ export class JSONLoader extends DataLoader {
...
 
         const mutation = new dgraph.Mutation();
         mutation.setSetNquads(`
-            uid(Property) <value> "${property.value}" .
-            uid(Property) <dgraph.type> "${property.type}" .
+            _:property <value> "${property.value}" .
+            _:property <dgraph.type> "${property.type}" .
 
-            uid(Entity) <properties> uid(Property) .
+            uid(Entity) <properties> _:property .
```
Comment (wlthomson) [inline unified-codes/tools/scripts/dgraph/migration/v2/DataLoader.ts:103]: 👍 .

File: unified-codes/tools/scripts/dgraph/migration/v2/DataParser.ts [hunk 3]
```diff
@@ -177,6 +185,8 @@ export class CSVParser extends DataParser {
               const key = column
                 .trim()
                 .toLowerCase()
+                .replace(/[\r\n]/g, '')
+                .replace(/(uc8) *\(([^)]*)\)/, '$1 $2')
```
Comment (wlthomson) [inline unified-codes/tools/scripts/dgraph/migration/v2/DataParser.ts:189]: Whoo, more regexes! A nice-to-have, but a potential justification for a helper class for spreadsheets, e.g. `column.getKey()` 😃 .

File: unified-codes/tools/scripts/dgraph/migration/v2/DataParser.ts [hunk 4]
```diff
@@ -223,28 +233,45 @@ export class CSVParser extends DataParser {
...
         } = row;
 
+        const productProperties = [];
+        const itemProperties = [];
+        productProperties.push({ type: 'code_rxnav', value: rxnav });
+        productProperties.push({ type: 'code_eml', value: who_eml_product });
+        itemProperties.push({ type: 'code_eml', value: who_eml_item });
+        productProperties.push({ type: 'code_nzulm', value: nzulm });
+        itemProperties.push({ type: 'code_nzulm', value: nzulm_item });
+        productProperties.push({ type: 'code_unspsc', value: unspsc });
```
Comment (wlthomson) [inline unified-codes/tools/scripts/dgraph/migration/v2/DataParser.ts:273]: Does `productProperties` etc. have a type here, e.g. `INode[]`?

**Review by wlthomson — APPROVED** [review]
LGTM!

File: unified-codes/tools/scripts/dgraph/migration/v2/DataParser.ts (hunk 3 above)
Comment (mark-prins) [reply unified-codes/tools/scripts/dgraph/migration/v2/DataParser.ts:189]: also noted in #311 so there were some constants added in that one.

File: unified-codes/tools/scripts/dgraph/migration/v2/DataParser.ts (hunk 4 above)
Comment (mark-prins) [reply unified-codes/tools/scripts/dgraph/migration/v2/DataParser.ts]: yeah, with a little tweaking they can. 4308b6a5c

**Pull request merged**